ibm/resource_ibm_private_dns_glb_monitor_test.go

```diff
@@ -0,0 +1,228 @@
+package ibm
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccIBMPrivateDNSGlbMonitor_Basic(t *testing.T) {
+	var resultprivatedns string
+	name := fmt.Sprintf("testpdnspn%s.com", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIBMPrivateDNSGlbMonitorDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMPrivateDNSGlbMonitorBasic(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMPrivateDNSGlbMonitorExists("ibm_dns_glb_monitor.test-pdns-monitor", resultprivatedns),
+					// dont check that specified values are set, this will be evident by lack of plan diff
+					// some values will get empty values
+					resource.TestCheckResourceAttr("ibm_dns_glb_monitor.test-pdns-monitor", "name", "test-pdns-glb-monitor"),
+					resource.TestCheckResourceAttr("ibm_dns_glb_monitor.test-pdns-monitor", "description", "Monitordescription"),
+					resource.TestCheckResourceAttr("ibm_dns_glb_monitor.test-pdns-monitor", "port", "80"), // default value
+				),
+			},
+			{
+				Config: testAccCheckIBMPrivateDNSGlbUpdateMonitorBasic(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMPrivateDNSGlbMonitorExists("ibm_dns_glb_monitor.test-pdns-monitor", resultprivatedns),
+					// dont check that specified values are set, this will be evident by lack of plan diff
+					// some values will get empty values
+					resource.TestCheckResourceAttr("ibm_dns_glb_monitor.test-pdns-monitor", "name", "test-pdns-glb-monitor-update"),
+					resource.TestCheckResourceAttr("ibm_dns_glb_monitor.test-pdns-monitor", "description", "UpdatedMonitordescription"),
+					resource.TestCheckResourceAttr("ibm_dns_glb_monitor.test-pdns-monitor", "port", "80"), // default value
+				),
+			},
+		},
+	})
+
+}
+
+func TestAccIBMPrivateDNSGlbMonitorImport(t *testing.T) {
+	var resultprivatedns string
+	name := fmt.Sprintf("testpdnszone%s.com", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIBMPrivateDNSGlbMonitorDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMPrivateDNSGlbMonitorBasic(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMPrivateDNSGlbMonitorExists("ibm_dns_glb_monitor.test-pdns-monitor", resultprivatedns),
+				),
+			},
+			{
+				ResourceName:      "ibm_dns_glb_monitor.test-pdns-monitor",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
```

Comment:
why are the attributes ignored here? Is there any specific reason?`ImportStateVerifyIgnore ` this should be used only for the parameters that are not set in READ method.. I see that `"headers", "expected_body"` are set in READ method

Reply:
removed